lib/std/special/compiler_rt/arm.zig

```diff
@@ -40,10 +41,28 @@ pub fn __aeabi_unwind_cpp_pr2() callconv(.C) void {}
 // This function can only clobber r0 according to the ABI
 pub fn __aeabi_read_tp() callconv(.Naked) void {
     @setRuntimeSafety(false);
-    asm volatile (
-        \\ mrc p15, 0, r0, c13, c0, 3
-        \\ bx lr
-    );
+
+    // for thumb2, ARMv2, ARMv3, ARMv4 and ARMv5 use the kuser_get_tls linux helper which is at address 0xffff0fe0
+    if (comptime builtin.cpu.arch.isThumb() and std.Target.arm.featureSetHas(builtin.cpu.features, .thumb2)) {
+        asm volatile (
```

Comment:
Should this be conditional on the OS?

Should we hoist the magic constant out and give it a name?